File: src/kangaroo_sys_ffi.mjs
// Small platform services: monotonic clock, environment access, exit.
// `Option$None$const` is the shared None instance; the `None` export is the
// class itself, which must not be returned as a value.
import { Option$None$const, Some } from "../gleam_stdlib/gleam/option.mjs";

export function now_ms() {
  return Date.now();
}

export function env(name) {
  if (typeof process !== "undefined" && process.env) {
    const value = process.env[name];
    return value === undefined ? Option$None$const : new Some(value);
  }
  return Option$None$const;
}

export function halt(code) {
  if (typeof process !== "undefined" && typeof process.exit === "function") {
    process.exit(code);
  }
  return undefined;
}

File: src/kangaroo_sys_ffi.erl
%% Small platform services used by the runner: monotonic clock, environment
%% access, and process exit.
-module(kangaroo_sys_ffi).
-export([now_ms/0, env/1, halt/1]).

now_ms() ->
    erlang:monotonic_time(millisecond).

env(Name) when is_binary(Name) ->
    case os:getenv(binary_to_list(Name)) of
        false -> none;
        Value -> {some, list_to_binary(Value)}
    end.

halt(Code) ->
    erlang:halt(Code).
